examples_tests/60.ClusteredRendering/main.cpp

```diff
@@ -1,4 +1,3831 @@
-int main()
+#define _NBL_STATIC_LIB
+#include <nabla.h>
+
+#include "../common/CommonAPI.h"
+#include "../common/Camera.hpp"
+#include "nbl/ext/ScreenShot/ScreenShot.h"
+#include "nbl/ui/ICursorControl.h"
+
+using namespace nbl;
+
+// #define SYNC_DEBUG
+#define DEBUG_VIZ
+
+#define CLIPMAP
+// #define OCTREE
+
+#define WG_DIM 256
+
+#define FATAL_LOG(x, ...) {logger->log(##x, system::ILogger::ELL_ERROR, __VA_ARGS__); exit(-1);}
+
+struct vec3
 {
-	return 0;
-}
\ No newline at end of file
+	float x, y, z;
+};
+
+struct alignas(16) vec3_aligned
+{
+	float x, y, z;
+};
+
+struct vec4
+{
+	float x, y, z, w;
+};
+
+struct uvec2
+{
+	uint32_t x, y;
+};
+
+struct nbl_glsl_ext_ClusteredLighting_SpotLight
+{
+	vec3 position;
+	float outerCosineOverCosineRange; // `cos(outerHalfAngle) / (cos(innerHalfAngle) - cos(outerHalfAngle))`
+	uvec2 intensity; // rgb19e7 encoded
+	uvec2 direction; // xyz encoded as packSNorm2x16, w used for storing `cosineRange`
+};
+
+struct nbl_glsl_shapes_AABB_t
+{
+	vec3_aligned minVx;
+	vec3_aligned maxVx;
+};
```

Comment:
this should all live in a `glsl`, then provide the `vec3`, vec3_aligned`, `vec4` etc. around a `#ifdef __cpluplus` like the others do